drivers/include/lm75.h

```diff
@@ -0,0 +1,209 @@
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+/**
+ * @defgroup    drivers_lm75     LM75 Temperature Sensor driver compile configuration
+ * @ingroup     drivers_sensors
+ * @brief       Driver for the lm75 temperature sensors.
+ */
+
+#ifndef LM75_H
+#define LM75_H
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name LM75 return values
+ */
+enum {
+    LM75_SUCCESS = 0,
+    LM75_ERROR_I2C,
+    LM75_ERROR
+};
+
+/**
+ * @brief temperature resolution structs of the LM75 sensors
+ */
+typedef struct {
+    uint16_t    os_res;     /**< resolution of the OS and HYST registers */
+    uint16_t    os_mult;    /**< multiplier required for getting the OS and HYST into ºC */
+    uint16_t    temp_res;   /**< resolution of the temperature register */
+    uint16_t    temp_mult;  /**< multiplier required for getting the temperature into ºC */
+    uint8_t     os_shift;   /**< how many bits need to be shifted (2 bytes - any unused bits) */
+    uint8_t     temp_shift;  /**< how many bits need to be shifted (2 bytes - any unused bits) */
+} lm75_temp_resolutions;
+
+/**
+ * @brief lm75a temperature resolutions
+ */
+static const lm75_temp_resolutions tmp1075_res = {
+    .os_res = 625,           /**< resolution in 0.0625ºC */
+    .os_mult = 10000,        /**< Must multiply by 10000 to get temp in ºC */
+    .os_shift = 4,           /**< Only the 12 most significant bits are needed */
+    .temp_res = 625,         /**< resolution in 0.0625ºC */
+    .temp_mult = 10000,      /**< Must multiply by 10000 to get temp in ºC */
+    .temp_shift = 4,         /**< Only the 12 most significant bits are needed */
+};
+
+/**
+ * @brief tmp1075 temperature resolutions
+ */
+static const lm75_temp_resolutions lm75a_res = {
+    .os_res = 5,        /**< resolution in 0.5ºC */
+    .os_mult = 10,      /**< Must multiply by 10 to get temp in ºC */
+    .os_shift = 7,      /**< Only the 9 most significant bits are needed */
+    .temp_res = 125,    /**< resolution in 0.125ºC */
+    .temp_mult = 1000,  /**< Must multiply by 1000 to get temp in ºC */
+    .temp_shift = 5,    /**< Only the 11 most significant bits are needed */
+};
+
+/**
+ * @brief params required for initialization
+ */
+typedef struct lm75_params {
+    i2c_t   i2c_bus;            /**< I2C Bus used */
+    lm75_temp_resolutions res;  /**< Temperature resolutions */
+    uint8_t i2c_addr;           /**< i2c address */
+    uint8_t shutdown_mode;      /**< Shutdown mode register */
+    uint8_t tm_mode;            /**< Thermistor Mode */
+    uint8_t polarity;           /**< OS polatiry register */
+    uint8_t fault_q;            /**< Fault Queue register */
+    uint8_t conv_rate;          /**< Conversion rate register - only in the tmp1075 sensor */
+    uint8_t one_shot_mode;      /**< One Shot Mode register - only in the tmp1075 sensor */
```

Comment:
Why are those registers mirrored here?

Reply:
They are required for the device initialization.